platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/style/BackgroundLayerTest.java

```diff
@@ -55,189 +58,249 @@ protected Class getActivityClass() {
+      }
+    });
   }
 
   @Test
   public void testBackgroundColorTransition() {
     validateTestSetup();
     setupLayer();
     Timber.i("background-colorTransitionOptions");
-    assertNotNull(layer);
-
-    // Set and Get
-    TransitionOptions options = new TransitionOptions(300, 100);
-    layer.setBackgroundColorTransition(options);
-    assertEquals(layer.getBackgroundColorTransition(), options);
+    invoke(mapboxMap, new MapboxMapAction.OnInvokeActionListener() {
+      @Override
+      public void onInvokeAction(UiController uiController, MapboxMap mapboxMap) {
+        assertNotNull(layer);
+
+        // Set and Get
+        TransitionOptions options = new TransitionOptions(300, 100);
+        layer.setBackgroundColorTransition(options);
+        assertEquals(layer.getBackgroundColorTransition(), options);
+      }
+    });
   }
 
   @Test
   public void testBackgroundColorAsConstant() {
     validateTestSetup();
     setupLayer();
     Timber.i("background-color");
-    assertNotNull(layer);
-
-    // Set and Get
-    layer.setProperties(backgroundColor("rgba(0, 0, 0, 1)"));
-    assertEquals((String) layer.getBackgroundColor().getValue(), (String) "rgba(0, 0, 0, 1)");
+    invoke(mapboxMap, new MapboxMapAction.OnInvokeActionListener() {
+      @Override
+      public void onInvokeAction(UiController uiController, MapboxMap mapboxMap) {
+        assertNotNull(layer);
+
+        // Set and Get
+        layer.setProperties(backgroundColor("rgba(0, 0, 0, 1)"));
+        assertEquals((String) layer.getBackgroundColor().getValue(), (String) "rgba(0, 0, 0, 1)");
+      }
+    });
   }
 
   @Test
   public void testBackgroundColorAsCameraFunction() {
     validateTestSetup();
     setupLayer();
     Timber.i("background-color");
-    assertNotNull(layer);
-
-    // Set
-    layer.setProperties(
-      backgroundColor(
-        zoom(
-          exponential(
-            stop(2, backgroundColor("rgba(0, 0, 0, 1)"))
-          ).withBase(0.5f)
-        )
-      )
-    );
-
-    // Verify
-    assertNotNull(layer.getBackgroundColor());
-    assertNotNull(layer.getBackgroundColor().getFunction());
-    assertEquals(CameraFunction.class, layer.getBackgroundColor().getFunction().getClass());
-    assertEquals(ExponentialStops.class, layer.getBackgroundColor().getFunction().getStops().getClass());
-    assertEquals(0.5f, ((ExponentialStops) layer.getBackgroundColor().getFunction().getStops()).getBase(), 0.001);
-    assertEquals(1, ((ExponentialStops) layer.getBackgroundColor().getFunction().getStops()).size());
+    invoke(mapboxMap, new MapboxMapAction.OnInvokeActionListener() {
+      @Override
+      public void onInvokeAction(UiController uiController, MapboxMap mapboxMap) {
+        assertNotNull(layer);
+
+        // Set
+        layer.setProperties(
+          backgroundColor(
+            zoom(
+                  exponential(
+                stop(2, backgroundColor("rgba(0, 0, 0, 1)"))
+              ).withBase(0.5f)
```

Comment:
Same thing happens in `CircleLayerTest.java`, `FillExtrusionLayerTest.java`, `FillLayerTest.java`, `LineLayerTest.java`, `RasterLayerTest.java` (generated classes).